submissions/api.py

```diff
@@ -684,6 +686,23 @@ def reset_score(student_id, course_id, item_id):
             item_id=item_id,
         )
 
+        if clear_state:
+            for sub in student_item.submission_set.all():
+                # soft-delete the Submission
+                sub.status = Submission.DELETED
+                sub.save(update_fields=["status"])
+
+                # Also clear out cached values
+                cache_key = "submissions.submission.{}".format(sub.uuid)
```

Comment:
Just to ensure we're clear - the individual submission cache (keyed on `submissions.submission.{}`) does not have a timeout, and needs to be handled here. The top scores cache is what would timeout, and I agree that it probably doesn't need to be explicitly handled here.